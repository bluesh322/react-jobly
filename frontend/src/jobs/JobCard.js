import React, {useState, useEffect, useContext} from "react";
import { Card, CardContent, Typography, Button, Box } from "@material-ui/core";
import UserContext from "../auth/UserContext";

const JobCard = ({ id, title, salary, equity, companyName }) => {
  const { hasAppliedForJob, applyForJob } = useContext(UserContext)
  const [applied, setApplied] = useState();

  useEffect(() => {
    setApplied(hasAppliedForJob(id));
  }, [id, hasAppliedForJob])

  const handleApply = async (evt) => {
    if (hasAppliedForJob(id)) return;
    applyForJob(id);
    setApplied(true);
  }

  return (
    <Box my={2}>
      <Card>
        <CardContent>
          <Typography>{title}</Typography>
          <Typography>{companyName}</Typography>
          {salary && (
            <Typography>Salary: {Number(salary).toLocaleString()}</Typography>
          )}
          <Typography>Equity: {equity}</Typography>
          <Button data-testid="apply" disabled={applied} onClick={handleApply} variant="contained" size="small" color="secondary">
            {applied ? "Applied" : "Apply"}
          </Button>
        </CardContent>
      </Card>
    </Box>
  );
};

export default JobCard;
