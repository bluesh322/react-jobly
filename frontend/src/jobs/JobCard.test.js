import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import TestUserProvider from "../auth/testUser";

test("renders without crashing", () => {
  render(
      <TestUserProvider>
        <JobCard></JobCard>
      </TestUserProvider>
  );
});

test("matches snapshot logged in", async () => {
  const { asFragment } = render(
      <TestUserProvider>
        <JobCard></JobCard>
      </TestUserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
