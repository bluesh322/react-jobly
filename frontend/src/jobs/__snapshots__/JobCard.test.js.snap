// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot logged in 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        />
        <p
          class="MuiTypography-root MuiTypography-body1"
        />
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Equity: 
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="apply"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Apply
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
